### ISSUE TITLE
iOS 14 Crash

### ISSUE THREAD

**Issue by normand1** [bazaarvoice/bv-ios-sdk#35]
## The Problem

This appears to be a bug specific to iOS 14. 
I'm seeing an issue with `BVMessageInterceptor` where `forwardingTargetForSelector:(SEL)aSelector` is invoked with the selector named: `_diffableDataSourceImpl`. This appears to trigger an infinite loop between `forwardingTargetForSelector:(SEL)aSelector` and `respondsToSelector:(SEL)aSelector`. 

I've found that adding a hardcoded check for `_diffableDataSourceImpl` fixes this issue:

```
- (BOOL)respondsToSelector:(SEL)aSelector {
    if ([@"_diffableDataSourceImpl" isEqualToString:NSStringFromSelector(aSelector)]) {
        return NO;
    }
...
```

I'd be interested to know if you too have observed this issue or if there is a fix that can be made on our app's side to make this crash no longer happen, or if there is a planned update for this library for iOS 14 if this is a known issue.

## Steps to Reproduce

Display a `BVReviewsCollectionView` and tap one of the collectionview cells in a simulator or device running iOS 14.

## Expected Behavior

The app should not crash

## Environment

* Specific BVSDK submodule that exhibits the issue: `BVMessageInterceptor`
* BVSDK version (or git revision) that exhibits the issue: `bv-ios-sdk 8.2.3`
* Last BVSDK version that did not exhibit the issue (if applicable):  `N/A`
* iOS version that exhibits the issue: `iOS14`
* Last iOS version that exhibits the issue (if applicable): `iOS14 (this appears to work fine on iOS13)`
* Xcode version: `Xcode 12`
* iOS Mobile SDK version build for under test: `Targeting iOS 10.0+`
* Real device or simulator: `Both`

## Details

See above

## Link to BVSDK Logs

`EXC_BAD_ACCESS` Error

## Code To Reproduce Issue [ Very Good To Have ]

See above

**Comment by balkrishna-bv** [comment]
Hi @normand1 ,

**Comment by balkrishna-bv** [comment]
Hi @normand1 ,
This issue has been fixed in version 8.4.1.
